python/ray/air/checkpoint.py

```diff
@@ -559,20 +559,25 @@ def _to_directory(self, path: str, move_instead_of_copy: bool = False) -> None:
             if local_path:
                 local_path_pathlib = Path(local_path).resolve()
                 if local_path_pathlib != path_pathlib:
-                    if path_pathlib.exists():
-                        shutil.rmtree(str(path_pathlib.absolute()))
                     # If this exists on the local path, just copy over
                     if move_instead_of_copy:
                         os.makedirs(str(path_pathlib.absolute()), exist_ok=True)
                         self._local_path = str(path_pathlib.absolute())
                         for inner in local_path_pathlib.iterdir():
+                            dest = path_pathlib / inner.name
+                            if dest.exists():
+                                # Ignore files that already exist.
+                                # For example, checkpoints from every rank may all have
+                                # a same .is_checkpoint file.
+                                continue
```

Comment:
Isn't it safer to always overwrite to ensure we have the most recent data?

Reply:
oh this is not data from different checkpoints.
this is data from differernt ranks of the same checkpoint.
since each rank all writes a .is_checkpoint file, it will conflict here if we don't ignore.
same below actually, when we copy things from different ranks.